omissis/go-jsonschema
# Stable output: Add some more names to anonymous Method classes

### Fix #54 
The root issue of the seemingly "random" output is the fact that the sorting method only sorts on items that have a `Name` decalred. Some `Method` structs didn't have this by default (such as the unmarshal and enum methods).

I'm not sure if this is a clean solution, but for me it generates solid and stable output so diffs between generated versions are clean :)

### What has been done
- Added some arbitrary names to some methods

### How to test
- Did 2 generations of slightly different schema files and compared in a diff

To debug, you can add the following code in line 110 in `pkg/codegen/model.go`  (see commit 96737c890f506b5f8c098a905b99a8ed7e3f4e6b):
```go

	for i, decl := range sorted {
		log.Printf("Sorted: %+v %+v", i, decl)
	}

```

this shows the sorted output.

### Todo
Maybe we should add automated tests?

EDIT:
Yep, just found the test data folder :clown_face: 
Looks like this will need a lot of restructuring in that testing data.. but first let me know if this is the right way to go!
